fendermint/app/src/cmd/genesis.rs

```diff
@@ -246,3 +252,64 @@ fn set_ipc_gateway(genesis_file: &PathBuf, args: &GenesisIpcGatewayArgs) -> anyh
         Ok(genesis)
     })
 }
+
+async fn new_child_genesis(
+    genesis_file: &PathBuf,
+    args: &GenesisIpcChildGenesisArgs,
+) -> anyhow::Result<()> {
+    // provider with the parent.
+    let parent_provider = IpcProvider::new_with_subnet(
+        None,
+        ipc_provider::config::Subnet {
+            id: args.subnet_id.clone(),
+            config: SubnetConfig::Fevm(EVMSubnet {
+                provider_http: args.parent_endpoint,
+                auth_token: None,
+                registry_addr: args.parent_registry,
+                gateway_addr: args.parent_gateway,
+            }),
+        },
+    )?;
+
+    let genesis_info = parent_provider.get_genesis_info(&args.subnet_id).await?;
+
+    // get gateway genesis
+    let ipc_params = ipc::IpcParams {
+        gateway: ipc::GatewayParams {
+            subnet_id: args.subnet_id.clone(),
+            bottom_up_check_period: genesis_info.bottom_up_checkpoint_period,
+            min_collateral: genesis_info.min_collateral,
+            msg_fee: genesis_info.msg_fee,
+            majority_percentage: genesis_info.majority_percentage,
+            active_validators_limit: genesis_info.active_validators_limit,
+        },
+    };
+    let genesis = Genesis {
+        // We set the genesis epoch as the genesis timestamp so it can be
+        // generated deterministically by all participants
+        // genesis_epoch should be a positive number, we can afford panicking
+        // here if this is not the case.
+        timestamp: Timestamp(genesis_info.genesis_epoch.try_into().unwrap()),
```

Comment:
Slightly unfortunate that we call it `epoch`, which is height in Filecoin. I try to avoid using that name in Fendermint for other stuff. What is `genesis_epoch` exactly?

Reply:
We cal it epoch because it comes directly from the state of the subnet actor in the parent, and we call it `epoch` in the actors. We could change it in the `IpcProvider` but it seems like a unnecessary overhead (unless you strongly think otherwise).

Reply:
Oh well. I wouldn't have chosen this contested name for the timestamp, but ho hum.